### PR TITLE
Force newlines for editorconfig aware editors

### PR DESCRIPTION
**Pull request by spuder**
Some editors (such as vscode) will honor a `.editorconfig` file in a git repo

This tells editorconfig aware editors to force a newline at the end of files

https://editorconfig.org/

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1 @@
+insert_final_newline = true
```
Comment (chrivers) [inline .editorconfig:1]: The line in the editor config, which is about inserting final newlines, it itself missing a final newline.

...the irony, sir, is *staggering*

Comment (spuder) [reply .editorconfig:1]: 🤔 Can this be done? Did I do it wrong? 🤔

Comment (spuder) [reply .editorconfig:2]: Looks like users need to have the editorconfig plugin enabled https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig

This makes this PR, only useful in certain situations and won't help everyone

Comment (chrivers) [reply .editorconfig]: Looks like most people will need a plugin - but also gitlab and github might pick this up on their own.. so.. Yeah, let's go :)

Comment (spuder) [reply .editorconfig]: Also added a .vscode/settings.json file for the vscode users who _dont_ have the editor config plugin enabled 
This should cover most of the vanilla setups. 

```
{
    "files.insertFinalNewline": true
}

```

Comment (chrivers) [reply .editorconfig]: LGTM, I'll rebase and merge

**Review by chrivers — APPROVED** [review]

**Pull request merged**